### PR TITLE
SAK-44379 Move file conversion service to session factory

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-44379

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java [hunk 1]
```diff
@@ -28,39 +28,47 @@
...
+
 @Slf4j
 @Setter
 public class FileConversionServiceImpl implements FileConversionService {
 
     @Autowired private ContentHostingService contentHostingService;
-    @Autowired private FileConversionQueueItemRepository repository;
     @Autowired private SecurityService securityService;
     @Autowired private ServerConfigurationService serverConfigurationService;
+    @Autowired private SqlService sqlService;
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:65]: You can do straight up sql using hibernate no need to bring in sqlservice

**Review by ern — CHANGES_REQUESTED** [review]
You will still want a DAO or Repository as this isolates the db ops from the service which means testing is easier to do.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java (hunk 1 above)
Comment (adrianfish) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:65]: I used sql service as I didn't want to load the entity before updating it with a simple update. Can you do that? Also, there's very little testing you can do with this. The main thing that needs testing is the actual libre office integration, ie: load testing. It seemed real overkill to create a Dao just for these simple calls when there'd be little to no yield in unit testing.

Comment (ern) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:65]: You can evict an Entity after making changes.

Comment (adrianfish) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java]: Changes made. All hibernate now. I still think it seems overkill to add a dao interface and impl, though. You can't even test this meaningfully without bringing up a docker container with libre in it.

**Comment by adrianfish** [comment]
@ern Can you take another look? Persistence layer now, and the test updated.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/persistence/FileConversionServiceRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private SessionFactory sessionFactory;
+
+    @Transactional
+    public List<FileConversionQueueItem> findByStatus(FileConversionQueueItem.Status status) {
+
+        Session session = sessionFactory.getCurrentSession();
+        List<FileConversionQueueItem> items
+            = session.createCriteria(FileConversionQueueItem.class)
+                .add(Restrictions.eq("status", status)).list();
+        session.close();
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/persistence/FileConversionServiceRepositoryImpl.java:31]: You don't need to manage the session as that is performed by Spring TX Manager

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/persistence/FileConversionServiceRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        Session session = sessionFactory.getCurrentSession();
+        FileConversionQueueItem mergedItem = null;
+        try {
+            mergedItem = (FileConversionQueueItem) session.merge(item);
+        } catch (TransientObjectException toe) {
+            session.persist(item);
+            mergedItem = item;
+        }
+        session.flush();
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/persistence/FileConversionServiceRepositoryImpl.java:46]: same here no need to manage the session

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java [hunk 4]
```diff
@@ -34,32 +34,30 @@
...
 import lombok.extern.slf4j.Slf4j;
 
+import javax.annotation.Resource;
+
 @Slf4j
 @Setter
 public class FileConversionServiceImpl implements FileConversionService {
 
     @Autowired private ContentHostingService contentHostingService;
-    @Autowired private FileConversionQueueItemRepository repository;
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:59]: I think this should be repository it's how it's done in other places

**Review by ern — CHANGES_REQUESTED** [review]
I would use the same pattern in assignments see SerializableRepository it mimics that of spring data and provides a common api and language.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/persistence/FileConversionServiceRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+public class FileConversionServiceRepositoryImpl extends SpringCrudRepositoryImpl<FileConversionQueueItem, Long>  implements FileConversionServiceRepository {
+
+    @Transactional
+    public List<FileConversionQueueItem> findByStatus(FileConversionQueueItem.Status status) {
+
+        Session session = sessionFactory.getCurrentSession();
+        List<FileConversionQueueItem> items
+            = session.createCriteria(FileConversionQueueItem.class)
+                .add(Restrictions.eq("status", status)).list();
+        session.close();
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/persistence/FileConversionServiceRepositoryImpl.java:23]:
```suggestion
```

**Review by ern — APPROVED** [review]
Nice looks much better, thx for the refactor

**Pull request merged**